job-service/src/main/java/com/hpe/caf/services/job/jobtype/JobType.java

```diff
@@ -81,21 +65,68 @@ public String getId() {
      */
     public WorkerAction buildTask(
         final String partitionId, final String jobId, final JsonNode parameters
-    ) throws BadRequestException, InvalidJobTypeDefinitionException {
-        final WorkerAction task = new WorkerAction();
-        task.setTaskClassifier(taskClassifier);
-        task.setTaskApiVersion(taskApiVersion);
-        task.setTaskPipe(taskPipe);
-        task.setTargetPipe(targetPipe);
-        final JsonNode taskData = taskDataBuilder.build(partitionId, jobId, parameters);
+    ) throws BadRequestException, InvalidJobTypeDefinitionException
+    {
+        final WorkerAction workerAction = new WorkerAction();
+        final JsonNode task = taskBuilder.build(partitionId, jobId, parameters);
         // job-put expects `Map` (or `String`), but only does a shallow type check
         try {
-            task.setTaskData(objectMapper.convertValue(taskData, Map.class));
+            // taskClassifier
+            final JsonNode taskClassifier = getNonNullPropertyFromTask(task, "taskClassifier");
+            if (!taskClassifier.isTextual() || taskClassifier.asText().isEmpty()) {
```

Comment:
Rather than doing all this manual validation, define a JSON Schema for the expected output format and include it as a resource.  Then at runtime validate the `task` json node against it.  This is much the same way that the parameters passed by the user are validated against the schema in the job type.